### DISCUSSION TITLE
Need help with Error: Export encountered errors on following paths: /buildrss

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#44024] [Help]
Hi there, I've got an error: 
`Error: Export encountered errors on following paths:
        /buildrss
    at E:\Projects\work\raiz\blog-nextjs\node_modules\next\dist\export\index.js:498:19
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Span.traceAsyncFn (path/blog-nextjs\node_modules\next\dist\trace\trace.js:75:20)
    at async path/blog-nextjs\node_modules\next\dist\build\index.js:1024:17
    at async Span.traceAsyncFn (path/blog-nextjs\node_modules\next\dist\trace\trace.js:75:20)
    at async path/blog-nextjs\node_modules\next\dist\build\index.js:898:13
    at async Span.traceAsyncFn (path/blog-nextjs\node_modules\next\dist\trace\trace.js:75:20)
    at async Object.build [as default] (path/blog-nextjs\node_modules\next\dist\build\index.js:82:25)`
    
    Error occurs after I've added a search page that looks like that:

export default function SearchPage(props) {
  const { query } = useRouter();
  const keyword = query.keyword;
  const { recentPosts, preview } = props;
  const [isContentLoading, setIsContentLoading] = useState(false);
  const [isErrorVisible, setIsErrorVisible] = useState(false);
  const [filteredPostSummaries, setFilteredPostSummaries] = useState({
    total: 0,
    items: [],
  });

  useEffect(() => {
    setIsContentLoading(true);
    async function initialFetch() {
      const result = await ContentfulApi.getPaginatedPostSummaries(1, keyword);
      if (result.total > 0) {
        setFilteredPostSummaries(result);
        setIsErrorVisible(false);
      } else setIsErrorVisible(true);
    }
    initialFetch();
    setIsContentLoading(false);
  }, [keyword]);

  return (
    <MainLayout preview={preview}>
      <PageMeta
        title={copy.searchPage.title}
        description={copy.searchPage.description}
        url={`${Config.pageMeta.blogIndex.url}/search?keyword=${keyword}`}
      />

      <ContentWrapper>
        <h3>{copy.searchPage.results}</h3>
        <div className={MainPage.container}>
          <SearchResultsContent
            isContentLoading={isContentLoading}
            postSummaries={filteredPostSummaries.items}
          />
          <AsideNavigationMenu
            isErrorVisible={isErrorVisible}
            recentPosts={recentPosts}
          />
        </div>
      </ContentWrapper>
    </MainLayout>
  );
}
// this contentful call, I'm using in any pages and it doesn't brake anything there
export async function getServerSideProps() {
  const recentPosts = await ContentfulApi.getRecentPostList();
  return {
    props: {
      preview: false,
      recentPosts,
    },
  };
}
`
And here is a function that I used to move to the search page:
` const handleSearchClick = (e) => {
    e.preventDefault();
    if (e.target.keyword.value !== "") {
      Router.push({
        pathname: "/search",
        query: { keyword: e.target.keyword.value },
      });
    }
  };
  `
Does anyone have any idea how I can solve that?

**Comment by icyJoseph** [comment]
Are you using `next export`? If so, then the issue is that `next export` produces a static HTML site. That means it is just HTML, CSS and JS files to be served, but it doesn't produce a server. Since there's no server various features are unavailable: https://nextjs.org/docs/advanced-features/static-html-export#unsupported-features - `GetServerSideProps` is one of them.

**Reply by ghost** [reply]
I'm not using it, It doesn't work even with `getStaticProps` (already changed it above). Even if I used `next export` it doesn't help.

**Reply by icyJoseph** [reply]
Nope using `next export` is not possible with `GSSP`. 

But since you are not using it then my next target is, although useEffect shouldn't really cause issues during build time or any server time really.

```tsx
useEffect(() => {
    setIsContentLoading(true);
    async function initialFetch() {
        const result = await ContentfulApi.getPaginatedPostSummaries(1, keyword);
        if (result.total > 0) {
            setFilteredPostSummaries(result);
            setIsErrorVisible(false);
        } else setIsErrorVisible(true);
    }
    initialFetch();
    setIsContentLoading(false);
}, [keyword]);

```

Where does `ContentfulAPI` come from? Is it supported on the client? 

Another target would be now that you've changed to `getStaticProps({ query })`, because there's no query during `getStaticProps` - it happens at build time, so there's no browser, nothing. 


> - params contains the route parameters for pages using [dynamic routes](https://nextjs.org/docs/routing/dynamic-routes). For example, if the page name is [id].js , then params will look like { id: ... }. You should use this together with getStaticPaths, which we’ll explain later.
> - preview is true if the page is in the [Preview Mode](https://nextjs.org/docs/advanced-features/preview-mode) and undefined otherwise.
> - previewData contains the [preview](https://nextjs.org/docs/advanced-features/preview-mode) data set by setPreviewData.
> - locale contains the active locale (if enabled).
> - locales contains all supported locales (if enabled).
> - defaultLocale contains the configured default locale (if enabled).

It seems to me that the error is in the data fetching function:

```tsx
export async function getServerSideProps(context) {
    try {
        const {keyword} = context.query;
        const recentPosts = await ContentfulApi.getRecentPostList();
        return {
            props: {
                preview: false,
                recentPosts,
                keyword,
            },
        };
    } catch (reason) {
        console.log(reason);
        throw reason; // let next.js fail, while we investigate
    }
}
```

Again, you've changed to `getStaticProps` and there's no query there... but then you'd need to wrap also the async work in a try catch and log the reason why it fails.

**Reply by ghost** [reply]
I update the code above without a query as props but get it from the router. It doesn't build anyway.
With your function, I getting a console error: 

GET http://localhost:3000/_next/static/webpack/5f56e585b1c4b8ba.webpack.hot-update.json 404 (Not Found)
__webpack_require__.hmrM @ webpack.js?ts=1671024804178:1129
Promise.then (async)
hotCheck @ webpack.js?ts=1671024804178:448
tryApplyUpdates @ hot-dev-client.js?1600:251
handleWarnings @ hot-dev-client.js?1600:95
processMessage @ hot-dev-client.js?1600:171
eval @ hot-dev-client.js?1600:29
eval @ websocket.js?a9be:58
handleMessage @ websocket.js?a9be:57

**Reply by icyJoseph** [reply]
Alright, with this much info it can really be anything. I am not sure how you are trying to build and that error is really unexpected. If you have a repository to share that'd be much better.

**Comment by reinaldo-vombo** [comment]
I had a similar error I'm using next 14, this is the page that causes the error `export default async function PhysicalAutonomyPage() {
   const content = await getTPhysicalAutonomyContent()
   const tableData = await getPhysicalAutonomyTableContent()
   const secondaryTable = await getPhysicalAutonomyTableByType()

   const data = tableData.map(item => ({
      IndicadoresChave: item.title,
      Total: item.total,
      Femenino: isGender(item.gender) ? item.gender.female : '',
      Masculino: isGender(item.gender) ? item.gender.male : '',
      ICF: item.icf,
      Data: item.startDate === null ? '// //' : [getYear(new Date(item.startDate)), getYear(new Date(item.endDate)),],
      Fonte: item.sorce,
   }))
   const data2 = secondaryTable.map((item) => ({
      IndicadoresSecundários: item.title,
      Total: item.total,
      Femenino: isGender(item.gender) ? item.gender.female : '',
      Masculino: isGender(item.gender) ? item.gender.male : '',
      ICF: item.icf,
      Data: item.startDate === null ? '// //' : [getYear(new Date(item.startDate)), getYear(new Date(item.endDate)),],
      Fonte: item.sorce,
   }));

   const ColumnDefs = [
      { field: 'IndicadoresChave', width: 400 },
      { field: 'Total' },
      { field: 'Femenino' },
      { field: 'Masculino' },
      { field: 'ICF' },
      { field: 'Data' },
      { field: 'Fonte', width: 600 },
   ];


   return <PhysicalAutonomy ColumnDefs={ColumnDefs} tabledata={data} tabledata2={data2} content1={content} />
}` and the PhysicalAutonomy component has all the htm of this page `'use client'

import Breadcrumbb from "@/components/Breadcrumbb"
import { TTableProps } from "../types"
import parse from 'html-react-parser';
import Maintable from "@/components/tables/Maintable";
import Modal from "@/components/Modal";
import PagesSkeleton from "@/components/PagesSkeleton";

const PhysicalAutonomy = ({ ColumnDefs, tabledata, tabledata2, content1, content2 }: TTableProps) => {
   return (
      <section>
         <Breadcrumbb pathName="Autonomia Física" subPath="indicadores" />
         <div className="padding">
            <div className="container">
               {content1 && content1.length > 0 ? (
                  <div className="space-y-7">
                     <h2 className="h2 mb-6 text-center">{content1[0].title}</h2>
                     <div className="text-justify text-[#212529] text-sm">
                        {parse(content1[0].description)}
                     </div>
                     <div className="space-y-3">
                        <Maintable content={tabledata} columnDefs={ColumnDefs} height={280} />
                        <Modal name='Indicadores Secundários' title='Indicadores Secundários' padding='bg-primary-black hover:bg-primary-red transition-colors p-2' className='w-full max-w-full max-h-full h-full'>
                           <Maintable content={tabledata2} columnDefs={ColumnDefs} height={500} />
                        </Modal>` and for some reason when I added 'use client' dir to this componet wich contain the html that fix error, perhaps its because im calling table component 'sorry about my ingles'
